go/toolchains.rst

```diff
@@ -18,22 +27,568 @@ customize the behavior of the core_ go rules.
 Design
 ------
 
-**TODO**: More information
+The go toolchain consists of two main layers, `the sdk`_ and `the toolchain`_.
+
+The SDK
+~~~~~~~
+
+At the bottom is the go sdk. This is the same thing you would get if you go to the main
+`go website`_ and download a `binary distribution`_.
+
+The go_sdk_ rule is responsible for downloading these, and adding just enough of a build file
+to expose the contents to Bazel. It currently also builds the cross compiled standard libraries
+for specific combinations, although we hope to make that an on demand step in the future.
+
+SDK's are specific to the host they are running on, and the version of go they want to use,
+but not the target they compile for. The go SDK is naturally `cross compiling`_.
+
+The go rules already adds a go_sdk_ rule for all the host and go version pairs that are shipped
+on the main go language website, so there should be no need to declare one of these unless you
+need a `forked version of go`_, however you may want to `control the version`_ or use the
+`installed sdk`_.
+
+The toolchain
+~~~~~~~~~~~~~
+
+Declaration
+^^^^^^^^^^^
+
+Toolchains are declared using the go_toolchain_ macro. This actually registers two Bazel
+toolchains, the main go toolchain, and a special bootstrap toolchain. The bootstrap toolchain
+is needed because the full toolchain includes tools that are compiled on demand and written in
+go, so we need a special cut down version of the toolchain to build those tools.
+
+Toolchains are pre-declared for all the known combinations of host, target and sdk, and the names
+are a predictable
+go\ **version**\ _\ **host**
```

Comment:
Some underscores are missing here.

Consider formatting this as `go_<version>_<host>` instead though. I don't think the bold emphasizes the substitution enough.